### ISSUE TITLE
[C#] - Invalid generated code for string field.

### ISSUE THREAD

**Issue by leandrogomes777** [real-logic/simple-binary-encoding#955]
Hi guys,

I'm using the SbeTools (I've tried the one in the nuget package and with the latest version here in git), but the generated code is invalid with the template i'm running. The DirectBuffer class don't have the signature method for the generated code **(Uint8Put)**.

The code generated below:
```
public void SetDeskID(string value)
{
    var encoding = DeskIDResolvedCharacterEncoding;
    const int sizeOfLengthField = 1;
    int limit = _parentMessage.Limit;
    int byteCount = _buffer.SetBytesFromString(encoding, value, limit + sizeOfLengthField);
    _parentMessage.Limit = limit + sizeOfLengthField + byteCount;
    _buffer.Uint8Put(limit, (ushort)byteCount);
}
```

The DirectBuffer code:

```
public unsafe void Uint8Put(int index, byte value)
{
    _pBuffer[index] = value;
}
```
Could you guys check?

Thanks.

**Comment by tmontgomery** [comment]
@JPWatson, what are your thoughts?

**Comment by IanCastro** [comment]
I have the same problem.
Looking the code, I found this line 'indent + INDENT + "_buffer.%3$sPut%4$s(limit, (ushort)byteCount);\n" +' on https://github.com/real-logic/simple-binary-encoding/blob/master/sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/csharp/CSharpGenerator.java
line 516
but when '%3$sPut%4$s' == Uint8Put this need change to: 'indent + INDENT + "_buffer.%3$sPut%4$s(limit, (byte)byteCount);\n" +'

**Comment by mjpt777** [comment]
@ZachBray Can you look at this?

**Comment by ZachBray** [comment]
I came across this a few weeks back as well. It is already fixed in #948: 
![image](https://github.com/real-logic/simple-binary-encoding/assets/2045239/f2ea4c2a-1031-44b0-843d-8ba8796e6305)
